### DISCUSSION TITLE
Replacing STM32F405RGT6 with STM32F405VGT6 for Extended RAM

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1492] [Q&A]
Hello everyone,

I am considering replacing the STM32F405RGT6 microcontroller in the Crazyflie with the STM32F405VGT6 to extend the RAM by utilizing external memory. Both microcontrollers belong to the same STM32F4 family, but the STM32F405VGT6 offers additional pins and features that can be leveraged for this purpose.

### Key Differences
- **STM32F405RGT6**: LQFP-64 package with 64 pins.
- **STM32F405VGT6**: LQFP-100 package with 100 pins.

### Goals
1. **Increase RAM**: Utilize external RAM for more complex applications.
2. **Maintain Firmware Compatibility**: Ensure the existing Crazyflie firmware can be adapted with minimal changes.

**Comment by knmcguire** [comment]
I don't see why not :D 

I missed the question in here though. Are you wondering if it is possible?

**Comment by amitx64** [comment]
@knmcguire, Yes, I wondered if it is possible or not. Does the firmware need any changes?

**Reply by knmcguire** [reply]
I'm sure that it will be possible, but if the firmware needs changes is difficult to say without trying it out. I'm sure that the makefile has some preconfiguration on what of type of stm it needs to be, but these are pretty similar. But I'm not the expert.

**Reply by amitx64** [reply]
@knmcguire Thank you
